stats.cpp

```diff
@@ -1,8 +1,105 @@
 #include "stats.h"
+#include <cmath>
 
-struct Stats compute_statistics(const float* numberset, int setlength) {
-    struct Stats s;
+EmailAlertSent Emailsent;
+LedAlertSent LedOn;
+
+Stats compute_statistics(const float* numberset, int setlength) {
+    Stats s;
     s.average = 0;
     s.min = 0;
     s.max = 0;
+    int i = 0;
+    float sum = 0.0;
+    if(numberset == NULL)
+    {
+        s.average = NAN;
+        s.min = NAN;
+        s.max = NAN;
+        return s;
+    }
+
+    if(0 == setlength%2)
```

Comment:
Why special handling of odd/even? Or is this a protection for zero-length numberset?